### PR TITLE
Changing minimum python version to 3.8 (was 3.9)

### PR DESCRIPTION
**Pull request by JosephCottam**
pyciemss has a minimum version of 3.8.  These changes allow more full pyciemss support.

### CODE REVIEW

**Review by djinnome — APPROVED** [review]
This looks 3.8 compliant to me.  I wonder if we should continue to support 3.8?

File: .github/workflows/python_tests.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.8, 3.9, '3.10']
```
Comment (djinnome) [inline .github/workflows/python_tests.yml:11]: Should '3.10' be in quotes or does it not matter?

Comment (JosephCottam) [reply .github/workflows/python_tests.yml:11]: It does not **seem** to matter.  I copied that bit out of the comparable pyciemss and wondered the same thing.

Comment (JosephCottam) [reply .github/workflows/python_tests.yml:11]: On second thought, i think its because 3.10 == 3.1 in numbers BUT we really mean a semantic version.

**Pull request merged**